src/test/java/uk/gov/pay/ledger/pact/PaymentCreatedEventQueueContractTest.java

```diff
@@ -57,19 +62,24 @@ public MessagePact createPaymentCreatedEventPact(MessagePactBuilder builder) {
     public void test() throws Exception {
         appRule.getSqsClient().sendMessage(SqsTestDocker.getQueueUrl("event-queue"), new String(currentMessage));
 
-        DatabaseTestHelper dbHelper = DatabaseTestHelper.aDatabaseTestHelper(appRule.getJdbi());
-        await().atMost(1, TimeUnit.SECONDS).until(
-                () -> {
-                    Map<String, Object> event;
-                    try {
-                        event = dbHelper.getEventByExternalId("externalId");
-                        return "externalId".equals(event.get("resource_external_id"));
+        TransactionDao transactionDao = new TransactionDao(appRule.getJdbi());
+        EventDao eventDao = appRule.getJdbi().onDemand(EventDao.class);
 
-                    } catch (NoSuchElementException e) {
-                        return false;
-                    }
-                }
+        await().atMost(1, TimeUnit.SECONDS).until(
+                () -> transactionDao.findTransactionByExternalId(externalId).isPresent()
         );
+
+        Optional<TransactionEntity> transaction = transactionDao.findTransactionByExternalId(externalId);
+        assertThat(transaction.isPresent(), is(true));
+        assertThat(transaction.get().getExternalId(), is(externalId));
+        assertThat(transaction.get().getAmount(), is(1000L));
+        assertThat(transaction.get().getDescription(), is("a description"));
+        assertThat(transaction.get().getReference(), is("aref"));
+        assertThat(transaction.get().getGatewayAccountId(), is(gatewayAccountId));
+        assertThat(transaction.get().getTotalAmount(), is(1069L));
+        assertThat(transaction.get().getTransactionDetails(), containsString("\"return_url\": \"https://example.org\""));
+        assertThat(transaction.get().getTransactionDetails(), containsString("\"payment_provider\": \"sandbox\""));
+        assertThat(transaction.get().getTransactionDetails(), containsString("\"corporate_surcharge\": 69"));
```

Comment:
Is `corporate_surcharge`/`total_amount` available for `PaymentCreated` event ? Or is it meant for `PaymentDetailsEntered` ?

Reply:
At the moment they're in both. I guess they should be cleaned up.

Reply:
Removed in https://github.com/alphagov/pay-ledger/compare/91af1595970716392e91434e438e36d98e73614c..30f130cd6ccd4868ee2d419bd46e23e2f3849c3b

Reply:
👍 Yeah it needs to be cleaned up